noir_stdlib/src/uint128.nr

```diff
@@ -112,23 +113,22 @@ impl U128 {
         }
     }
 
-    pub fn from_integer<T>(i: T) -> U128 {
-        let f = crate::as_field(i);
+    pub fn from_field(f: Field) -> U128 {
         // Reject values which would overflow a u128
         f.assert_max_bit_size(128);
         let lo = f as u64 as Field;
-        let hi = (f - lo) / pow64;
+        let hi = (f - lo) / POW_64;
         U128 { lo, hi }
     }
 
     pub fn to_integer<T>(self) -> T {
-        crate::from_field(self.lo + self.hi * pow64)
+        crate::from_field(self.lo + self.hi * POW_64)
```

Comment:
Just saw https://github.com/noir-lang/noir/pull/4944

Reply:
I've kept `from_field` and not implemented `From<Field>` as this conversion can fail.

Reply:
I think we may want to have something along the lines of an `AssertFrom` trait which is essentially `From` but will throw a constraint error if the conversion isn't valid.

Reply:
Ah, I was just thinking of `TryFrom`

Reply:
How would `TryFrom` work here? I'm thinking of `AssertFrom` as we have a number of conversions where it's much easier to assert that the conversion is valid rather than checking whether it is valid or not.

Reply:
`TryFrom` would always be similar to `AssertFrom`, it just gives users the option to recover from an error if possible but would be less efficient than the AssertFrom in cases where users do just want to panic on failure.